dlt/normalize/items_normalizers.py

```diff
@@ -52,6 +65,267 @@ def __init__(
     def __call__(self, extracted_items_file: str, root_table_name: str) -> List[TSchemaUpdate]: ...
 
 
+class ModelItemsNormalizer(ItemsNormalizer):
+    def _normalize_casefold(self, ident: str) -> str:
+        return self.config.destination_capabilities.casefold_identifier(
+            self.schema.naming.normalize_identifier(ident)
+        )
+
+    def _uuid_expr_for_dialect(self, dialect: str) -> sqlglot.exp.Expression:
+        """
+        Generates a UUID expression based on the specified dialect.
+
+        Args:
+            dialect (str): The SQL dialect for which the UUID expression needs to be generated.
+
+        Returns:
+            sqlglot.exp.Expression: A SQL expression that generates a UUID for the specified dialect.
+        """
+        # NOTE: redshift and sqlite don't have an in-built uuid function
+        if dialect == "redshift":
+            row_num = sqlglot.exp.Window(
+                this=sqlglot.exp.Anonymous(this="row_number"),
```

Comment:
I have a question about this: Is there some possibility of this method producing duplicates? Is this row_number the row_number of the current insert, or of the whole table? In both cases, you can produce scenarios where 2 rows will end up with the same row_number. In case one it will happen directly on the next load if this row_number starts at 0 again, and in case two it will happen if rows are for any reason deleted and you load again. 

Can we add a random seed (maybe the load id) to the row number so that this does not happen?

Reply:
For redshift there also is this method: https://stackoverflow.com/questions/46578889/how-to-generate-12-digit-unique-number-in-redshift. I'm not sure how feasible this is though...